compute_endpoint/tests/unit/test_endpoint_config.py

```diff
@@ -0,0 +1,33 @@
+import typing as t
+
+import pytest
+from globus_compute_endpoint.endpoint.config.model import ConfigModel
+
+
+@pytest.fixture
+def config_dict():
+    return {"engine": {"type": "HighThroughputEngine"}}
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        ("worker_ports", (50000, 55000)),
+        ("worker_port_range", (50000, 55000)),
+        ("interchange_port_range", (50000, 55000)),
+    ],
+)
+def test_config_model_tuple_conversions(config_dict: dict, data: t.Tuple[str, t.Tuple]):
+    field, expected_val = data
+
+    config_dict["engine"][field] = expected_val
+    model = ConfigModel(**config_dict)
+    assert getattr(model.engine, field) == expected_val
```

Comment:
Ought we check non-tuple types as well?  As in, what the `yaml.safe_load()` returns à la the initial report?

An additional couple of tests might also be to load a YAML file with this user-found expectation.  (In particular, I don't see that we have verified the `yaml_load()` -> `ConfigDict()` route so well in our tests.)

Reply:
This test includes checks for passing in tuples, lists, and integers.

Reply:
Yep; was going too fast earlier, apparently.  I see it now ... *just below*.  :facepalm: